deploy/runtools/runtime_config.py

```diff
@@ -40,7 +40,10 @@ class RuntimeHWConfig:
 
     # TODO: should be abstracted out between platforms with a URI
     agfi: Optional[str]
+    """User-specified, possibly-URI, path to xclbin"""
```

Comment:
IMO we should just enforce that this is a URI (i.e. enforce `file:///path/to/file`)

Reply:
Probably.  The only thing we would be losing is the ability for someone to put a relative path in their hwdb (because file URIs need the full path).

While on this topic, I wonder how fsspec would handle `file://<ipaddress>/path/to/file` because that is a valid file URI.  https://en.wikipedia.org/wiki/File_URI_scheme. I had forgotten that you can give a hostname in a file URI because I never do.

Reply:
I don't think that having a relative path is that important. I think everything using a URI would be clearer to users.

As for having a hostname in a URI, I assume it probably just does a `rsync` like operation to retrieve the file. Whats your concern?

Reply:
authentication will be annoying depending on how it's implemented under the hood. 

I personally would like relative paths to be supported.

Reply:
> Whats your concern?

That we would need to document it because users will need to know how to setup their authentication.

Reply:
If relative paths are supported, I think it would be nice to enforce that full paths are done using URIs while relative paths shouldn't (maybe `pathlib` has a way to enforce a `Path` is a relative path)? However, this isn't blocking so if that can't happen / is too much effort we can just do what currently exists.

Reply:
It wouldn't be hard to `raise` something if `localize_xclbin()` is given an absolute filepath.  I'm concerned that it would be better for this to be caught earlier like when the config is read and validated against a schema.

Reply:
@abejgonzalez, the more I think about this, the less I feel like enforcing where users use a `file://` URI vs a simple string path.  Please speak up if you feel strongly enough about this that it should block merging.

Reply:
@abejgonzalez , since you requested changes, I think I need you to approve.  I guess there are new mypy errors I need to cleanup also.